### PR TITLE
Early draft of log_artifact

### PR DESCRIPTION
**Pull request by daavoo**

### CODE REVIEW

File: src/dvclive/live.py [hunk 1]
```diff
@@ -299,6 +301,19 @@ def log_param(
         """Saves the given parameter value to yaml"""
         self.log_params({name: val})
 
+    def log_artifact(
+        self,
+        path: str,
+    ):
+        """Saves the given parameter value to yaml"""
+        if self._dvc_repo is not None and not self._inside_dvc_exp:
```
Comment (daavoo) [inline src/dvclive/live.py:309]: We are going to encounter lock issues here if we are inside `dvc repro` execution. 
Probably need to add a new property `self._inside_dvc_repro` and a way to infer this.

**Comment by dberenbaum** [comment]
@daavoo Can you discuss with @karajan1001? He was going to take this issue.

**Comment by daavoo** [comment]
> @daavoo Can you discuss with @karajan1001? He was going to take this issue.

Yes, we discussed and I open the P.R. to share what I already had

File: src/dvclive/live.py [hunk 2]
```diff
@@ -299,6 +301,19 @@ def log_param(
...
+            dvc_file = self._dvc_repo.add(path)[0].addressing
```
Comment (dberenbaum) [inline src/dvclive/live.py:310]: Should we check if it already exists as an output in either `dvc.yaml` or `.dvc`?

File: src/dvclive/live.py [hunk 3]
```diff
@@ -299,6 +301,19 @@ def log_param(
...
+            if self._save_dvc_exp:
```
Comment (dberenbaum) [inline src/dvclive/live.py:311]: Any reason not to do this regardless of `self._save_dvc_exp`?

File: src/dvclive/live.py (hunk 2 above)
Comment (karajan1001) [reply src/dvclive/live.py:310]: I think we can wrap the exception here.

**Comment by daavoo** [comment]
Closing in favor of https://github.com/iterative/dvclive/pull/464

**Pull request closed without merging**